packages/wonder-blocks-data/util/response-cache.test.js

```diff
@@ -111,117 +112,597 @@ describe("./response-cache.js", () => {
     });
 
     describe("#cacheData", () => {
-        it("should add the entry to the cache", () => {
+        describe("no custom cache", () => {
+            it("should add the entry to the cache", () => {
+                // Arrange
+                const cache = new ResponseCache();
+                const fakeHandler: IRequestHandler<string, string> = {
+                    getKey: () => "MY_KEY",
+                    type: "MY_HANDLER",
+                    shouldRefreshCache: () => false,
+                    fulfillRequest: jest.fn(),
+                    cache: null,
+                };
+
+                // Act
+                cache.cacheData(fakeHandler, "options", "data");
+                const result = cache.getEntry(fakeHandler, "options");
+
+                // Assert
+                expect(result).toStrictEqual({data: "data"});
+            });
+
+            it("should replace the entry in the handler subcache", () => {
+                // Arrange
+                const cache = new ResponseCache({
+                    MY_HANDLER: {
+                        MY_KEY: {error: "Oh no!"},
+                    },
+                });
+                const fakeHandler: IRequestHandler<string, string> = {
+                    getKey: () => "MY_KEY",
+                    type: "MY_HANDLER",
+                    shouldRefreshCache: () => false,
+                    fulfillRequest: jest.fn(),
+                    cache: null,
+                };
+
+                // Act
+                cache.cacheData(fakeHandler, "options", "other_data");
+                const result = cache.getEntry(fakeHandler, "options");
+
+                // Assert
+                expect(result).toStrictEqual({data: "other_data"});
+            });
+        });
+
+        describe("with custom cache", () => {
+            it("SSR should use the framework cache, not the custom cache", () => {
+                // Arrange
+                jest.spyOn(Server, "isServerSide").mockReturnValue(true);
+                const cache = new ResponseCache();
+                const customCache = {
+                    store: jest.fn(),
+                    retrieve: jest.fn(),
```

Comment:
`get`/`set` seem like more common names.  These are fine too.  No action required.

Reply:
I don't feel that `get`/`set` really corresponds to caching. It sounds much more like just retrieving properties. I wanted to codify that it was doing work of some kind.